From dc9b80f225164a8d412476d8f5b50033f2802315 Mon Sep 17 00:00:00 2001
From: Otavio Pontes <otavio.pontes@intel.com>
Date: Tue, 19 Apr 2016 15:37:46 -0300
Subject: [PATCH 50/50] oic-gen: Always generate code using same order of
 resource types

os.listdir documentation states that list order is arbitrary. Sorting
the list dir to make sure we are always generating the oic file in same
order.

Signed-off-by: Otavio Pontes <otavio.pontes@intel.com>
---
 data/scripts/sol-oic-gen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/scripts/sol-oic-gen.py b/data/scripts/sol-oic-gen.py
index c03e2e6..c8523a8 100755
--- a/data/scripts/sol-oic-gen.py
+++ b/data/scripts/sol-oic-gen.py
@@ -1711,7 +1711,7 @@ if __name__ == '__main__':
     generated = []
     print('Generating code for schemas: ', end='')
     for schema_dir in args.schema_dirs:
-        for path in (f for f in os.listdir(schema_dir) if seems_schema(f)):
+        for path in (f for f in sorted(os.listdir(schema_dir)) if seems_schema(f)):
             print(path, end=', ')
 
             try:
-- 
2.4.11

